internal/backend/post.go

```diff
@@ -92,6 +137,46 @@ func (d *delphisBackend) GetPostsByDiscussionID(ctx context.Context, discussionI
 	//return d.db.GetPostsByDiscussionID(ctx, discussionID)
 }
 
+func (d *delphisBackend) GetMentionedEntities(ctx context.Context, entityIDs []string) ([]model.Entity, error) {
+	var entities []model.Entity
+	var participantIDs []string
+	var discussionIDs []string
+
+	// Iterate over mentioned entities and divide into participants and discussions
+	for _, entityID := range entityIDs {
+		s := strings.Split(entityID, ":")
+		if s[0] == model.ParticipantPrefix {
+			participantIDs = append(participantIDs, s[1])
+		} else if s[0] == model.DiscussionPrefix {
+			discussionIDs = append(discussionIDs, s[1])
+
+		} else {
```

Comment:
We need to consider what the ramifications are for the post itself. Is it doable if this field is non existent.

My knee jerk is to say, at the very least, we should put `nil` for this entry and either let the client barf or not. Having this not exist will cause an entire off by one error which is far worse.

Reply:
Like this is the reason maps are better than arrays for this type of thing (see comment below where you ask about maps).